src/AzureClient/Magic/ConnectMagic.cs

```diff
@@ -93,16 +103,41 @@ public ConnectMagic(IAzureClient azureClient)
         public override async Task<ExecutionResult> RunAsync(string input, IChannel channel)
         {
             var inputParameters = ParseInputParameters(input);
-
-            var storageAccountConnectionString = inputParameters.DecodeParameter<string>(ParameterNameStorageAccountConnectionString);
-            if (string.IsNullOrEmpty(storageAccountConnectionString))
+            if (!inputParameters.Any())
             {
                 return await AzureClient.GetConnectionStatusAsync(channel);
             }
 
-            var subscriptionId = inputParameters.DecodeParameter<string>(ParameterNameSubscriptionId);
-            var resourceGroupName = inputParameters.DecodeParameter<string>(ParameterNameResourceGroupName);
-            var workspaceName = inputParameters.DecodeParameter<string>(ParameterNameWorkspaceName);
+            var resourceId = inputParameters.DecodeParameter<string>(ParameterNameResourceId, defaultValue: string.Empty);
+            var subscriptionId = string.Empty;
+            var resourceGroupName = string.Empty;
+            var workspaceName = string.Empty;
+
+            // A valid resource ID looks like:
+            // /subscriptions/f846b2bd-d0e2-4a1d-8141-4c6944a9d387/resourceGroups/RESOURCE_GROUP_NAME/providers/Microsoft.Quantum/Workspaces/WORKSPACE_NAME
+            var resourceIdRegex = new Regex(
+                @"^\/subscriptions\/([a-zA-Z0-9\-]*)\/resourceGroups\/([^\s\/]*)\/providers\/Microsoft\.Quantum\/Workspaces\/([^\s\/]*)$");
+            if (!string.IsNullOrEmpty(resourceId) && resourceIdRegex.IsMatch(resourceId))
```

Comment:
If `resourceId` is never null (which it looks like might be true, since `DecodeParameter` has `defaultValue: string.Empty`), you could do this to avoid matching the regex twice:

```suggestion
            var match = resourceIdRegex.Match(resourceId);
            if (match.Success)
```

You could also do `Match(resourceId ?? string.Empty)` if `resourceId` can be null.

Reply:
You're right, `resourceId` cannot be null here. Applying this suggestion, thanks!